examples/sensor-channels/src/components/SensorDataHandler.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from "react";
+
+import { SensorData } from "./SensorData";
+
+function getLocaleFormat(timestamp) {
+  const event = new Date(timestamp);
+  const locale = window.navigator.language;
+  return event.toLocaleString(locale);
+}
+
+class SensorDataHandler extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      sensors: {},
+      device: {
+        connection_type: false
+      },
+      connected: false,
+      alerts: []
+    };
+  }
+
+  setDeviceStatus = () => {
+    const { device, astarte } = this.props;
+    astarte.getDeviceDataById(device).then(response => {
+      this.setState({ device: { connection_type: response.connected } });
+    });
+  };
+
+  onInComingData = response => {
+    if (response.event === "new_event") {
+      const { sensors } = this.state;
+      const time = response.payload.timestamp;
+      const event = response.payload.event;
+      if (event.path) {
+        sensors[event.path] = {
+          value: event.value,
+          // event.path => '/${device_name}/value'
+          // name = event.path.split("/")[1]
+          name: event.path.split("/")[1],
+          timestamp: getLocaleFormat(time)
+        };
+      }
+      this.setState({ sensors });
+    }
+  };
+
+  onSocketError = () => {
+    const { alerts } = this.state;
+    alerts.unshift({ type: "info", msg: "An error occurred" });
```

Comment:
It would be nice to see which error occurred. Also, you can use warning styling. 'Info` is not really suited for errors

Reply:
for print error, I need to find a way to call this on socket error method, because socket always connected so not calling this method

Reply:
the docs say that onError function callback accepts 1 argument https://hexdocs.pm/phoenix/js/#socket
Try hardcoding a wrong URL and see what's in the error parameter